src/app/charts/candlestick-chart/candlestick-chart-view/candlestick-chart-view.component.ts

```diff
@@ -0,0 +1,250 @@
+            .style('text-anchor', 'middle')
+            .style('font-weight', 'bold')
+            .text('PHENOTYPE');
+
+        // Draw Y axis
+        const y = d3.scaleLinear()
+            .domain([this.minValue, this.maxValue])
+            .range([height, 0]);
+
+        group.append('g')
+            .attr('class', 'axis y-axis')
+            .call(d3.axisLeft(y));
+
+        // Draw Y axis title
+        group.append('text')
+            .attr('transform', `rotate(-90) translate(${-height / 2}, ${-margin.left / 2})`)
+            .style('text-anchor', 'middle')
+            .style('font-weight', 'bold')
+            .text('ODDS RATIO');
+
+        // Draw vertical lines
+        group.selectAll('.vertLines')
+            .data(this.chartData)
+            .enter()
+            .append('line')
+            .attr('class', 'vertLines')
+            .attr('x1', d => x(d.key))
+            .attr('x2', d => x(d.key))
+            .attr('y1', d => y(d.value.min))
+            .attr('y2', d => y(d.value.max))
+            .attr('stroke', color)
+            .attr('stroke-width', 1.5);
+
+        // Draw mid circle (mean value)
+        group.selectAll('.meanCircle')
+            .data(this.chartData)
+            .enter()
+            .append('circle')
+            .attr('class', 'meanCircle')
+            .attr('cx', d => x(d.key))
+            .attr('cy', d => y(d.value.mean))
+            .attr('r', 9)
+            .attr('stroke', color)
+            .style('fill', color)
+            .on('mouseover', d => {
+                const isOrNot = d.value.mean > 1.0 ? 'is' : 'is not';
+                const msg = `${this.hgicId} ${isOrNot} significantly correlated with ${d.key}, with an odds ratio of ${d.value.mean} and an adjusted p-value of ${d.value.pval_adj}.`;
+                this.showTooltip(this.tooltip, msg);
+            })
+            .on('mouseout', d => this.hideTooltip(this.tooltip));
+
+        // Draw X-axis tooltip
+        group.selectAll('.x-axis .tick text')
+            .data(this.chartData)
+            .on('mouseover', d => {
+                const msg = this.chartService.getTooltipText(d.key);
+                if (msg) {
+                    this.showTooltip(this.xAxisTooltip, msg);
+                }
+            })
+            .on('mouseout', d => this.hideTooltip(this.xAxisTooltip));
+
+    }
+
+    createTooltip(name) {
+        return d3.select('body').append('div')
+            .attr('class', `${name}`)
+            .style('opacity', 0);
+    }
+
+    showTooltip(tooltip, msg) {
+        tooltip.transition()
+            .duration(200)
+            .text(msg)
+            .style('opacity', 1)
+            .style('left', (d3.event.pageX) + 'px')
+            .style('top', (d3.event.pageY + 30) + 'px');
+    }
+
+    hideTooltip(tooltip) {
+        tooltip.transition()
+            .duration(500)
+            .style('opacity', 0);
+    }
+
+    // Helper functions for testing
```

Comment:
I don't think there should be code in the component with the sole purpose of just facilitating testing

Reply:
Yes, I can remove the private keyword and my test should be able to run without these helper functions.